architecture-decision-records/0008-keycloak-user-administrators-configuration.md

```diff
@@ -0,0 +1,104 @@
+# 8. Keycloak User Administrators Configuration
+
+**Date:** 2020-06-17
+
+## Context
+
+Keycloak provides a flexible way to organise access for users to its administration functions.
+
+TDR requires user configuration that allows the following:
+* Set of users permission to manage transferring body users, ie the users of the TDR application.
+* Set of users permission to manage keycloak users, ie those internal TNA users who can administer Keycloak itself, including setting up TDR application users.
+
+The configuration needs to consider the following principles:
+* *Least privilege*: only allow the user the minimum level of permissions to perform the requires operations
+* *Elevated privileges*: prevent a user accruing additional permissions above what they require
+* *User data protection*: ensure user data is secure
+
+## Decision
+
+Decided to define user administrators in the "master" realm. 
+
+There is some benefit from having the user administrators in the master realm as such users will not have access to themselves and cannot give themselves elevated privileges.
```

Comment:
I'm still not sure about this argument - if an admin account was compromised by an attacker, what extra damage could they do under Option 1 vs Option 2? It's counterintuitive that they can edit their own user, but I don't think moving them to a higher realm limits what they could actually do to the rest of the system. But I'm curious to know if I've missed something.

Like I said in the discussion the other day - I agree with the decision, but I think your point on line 24 is the main reason if there isn't any difference between the options in terms of TDR.